deploy/terraform/modules/hdb_node/infrastructure.tf

```
@@ -36,16 +36,16 @@ data "azurerm_subnet" "subnet-sap-db" {
 resource "azurerm_network_security_group" "nsg-admin" {
   count               = local.enable_deployment ? (local.sub_admin_nsg_exists ? 0 : 1) : 0
   name                = local.sub_admin_nsg_name
-  location            = var.resource-group[0].location
```

location is missing:
```suggestion
  virtual_network_name = var.vnet-sap[0].location
```

---

fixed